### PR TITLE
Filter out cells in SeuratV4 integration

### PR DESCRIPTION
**Pull request by gerbeldo**
# Description
Cells were not filtered when running seuratv4 integration, this PR fixes it.

# Details
#### URL to issue
N/A

#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R [hunk 1]
```diff
@@ -86,6 +81,27 @@ run_seuratv4 <- function(scdata_list, config, cells_id) {
 }
 
 
+#' prepare scdata list for seurat integration
+#'
+#' preprocess the scdata list before integration>
```
Comment (saracastel) [inline pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R:86]: just a very small thing: remove the extra ">" ?

**Review by saracastel — APPROVED** [review]

File: pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R [hunk 2]
```diff
@@ -31,18 +31,13 @@ run_seuratv4 <- function(scdata_list, config, cells_id) {
...
   for (i in 1:length(scdata_list)) {
     # we need RNA assay to compute the integrated matrix
     Seurat::DefaultAssay(scdata_list[[i]]) <- "RNA"
 
-    # remove cell cycle genes if needed
-    if (length(exclude_groups) > 0) {
-      scdata_list[[i]] <- remove_genes(scdata_list[[i]], exclude_groups)
-    }
-
     if (normalization == "LogNormalize") {
```
Comment (ogibson) [inline pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R:41]: Shouldn't the defaultassay <- RNA be a part of the prepare list function?

Do we even need the defaultAssay <- RNA?

**Review by ogibson — APPROVED** [review]

File: pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R (hunk 1 above)
Comment (gerbeldo) [inline pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R:86]:
```suggestion
#' preprocess the scdata list before integration
```

File: pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R (hunk 2 above)
Comment (gerbeldo) [reply pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R:41]: not sure if we need it. I followed the same structure as harmony integration.

Comment (ogibson) [reply pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R:41]: My concern is that remove_genes subsets:

 scdata <- subset(scdata, features = keep_genes)

If we need to fix the defaultAssay to RNA, that means that this subset might *not* be applied on the RNA assay at this stage.

**Pull request merged**